File: SuboptimalProcedureLearner/SuboptimalProcedureLearner/Misc/Deepcopy.cs
using System;
using System.Collections.Generic;

namespace Misc
{
    class Deepcopy
    {
        // TODO< too slow, speed it up with C# magic >
        public static List<int> deepCopyListInt(List<int> input)
        {
            List<int> result;

            result = new List<int>();

            foreach( int iteration in input )
            {
                result.Add(iteration);
            }

            return result;
        }
    }
}

File: src/java/ptrman/agi-x0/src/java/ptrman/misc/Deepcopy.java
package ptrman.misc;

import java.util.ArrayList;
import java.util.List;

public class Deepcopy {
    public static<Type> List<Type> deepCopyList(List<Type> input) {
        List<Type> result = new ArrayList<>();
        for( Type __dummyForeachVar0 : input ) {
            result.add(__dummyForeachVar0);
        }
        return result;
    }
}

